### DISCUSSION TITLE
aromatic sulfur in sulfoxide  is not counted as aromatic

### DISCUSSION THREAD

**Discussion by cottonkiet** [rdkit/rdkit#5255] [Q&A]
Hi all,
For an aromatic sulfoxide: O=S1C2=C(C=CC=C2)C2=C1C=CC=C2
It seems that the sulfur is not counted as an aromatic atom by rdkit. Is this right? 

```
mol = Chem.MolFromSmiles('O=S1C2=C(C=CC=C2)C2=C1C=CC=C2')
aromatic_carbon = Chem.MolFromSmarts('s')
print(mol.GetSubstructMatches(aromatic_carbon))
len(list(mol.GetAromaticAtoms()))
```

RDKit version: 2021.09.5
OS: Windows
Python version (if relevant): 3.9
Are you using conda? Yes
If you are using conda, which channel did you install the rdkit from? default

**Comment by wopozka** [comment]
The basic question is whether this sulfur after oxidation is yet aromatic? According to literature data (eg, doi 10.3184/030823402103172167) thiophenes after oxidation exhibit much lower aromaticity and readily undergo Diels-Alder reaction. The later suggest that we have diene rather than aromatic delocalization.

**Reply by cottonkiet** [reply]
Thanks a lot! I did not realize that rdkit's aromaticity model takes into the experimental aromaticity information besides formal aromaticity (4n+2) rule. Do you happen to known the place where we can have the full details of the aromaticity model by rdkit? I find the 'The RDKit Aromaticity Model' chapter in the 'The RDKit Book' but it only it only has examples (not a full list) of the aromatic fragment. Thanks!

**Reply by wopozka** [reply]
@cottonkiet 
I do not know how it is implemented internaly (I did not try to analyse source code), maybe it considers only nonoxidized thiophenes (and selenophenes) as aromatic. What you can do is to select aromaticity model AFAIR, and then simple 4n+2 rule will be used. In general aromaticity is complicated problem in cheminformatics, and it might not work as you excpect in some cases. Therefore you should be rather ready to test whether given molecule is aromatic (in rdkit point of view) and to act accordingly, than expect correction in algorithm. One of my favroite example is this molecule c1coc2cccc-2c1 - look, all structure is aromatic, but one bond between two aromatic carbons is considered nonaromatic. The same applies for azulenes (c1ccc2cccc-2cc1). All these examples are limitation of aromaticity model.

**Reply by cottonkiet** [reply]
Thanks a lot for helping me to understand the rdkit more!
